### ISSUE TITLE
netcoreapp11 methods did not get exposed in System.Runtime package

### ISSUE THREAD

**Issue by adamsitnik** [dotnet/corefx#13717]
Today when .NET Core 1.1 got released I wanted to make use of the new `GC.GetAllocatedBytesForCurrentThread ` method which got exposed [here](https://github.com/dotnet/corefx/pull/12489) by @jkotas .

However I can't do so, I get `error CS0117: 'GC' does not contain a definition for 'GetAllocatedBytesForCurrentThread'` error when trying to use it.

`dotnet --version: 1.0.0-preview2-1-003155`

My project.json file:
```json
{
  "version": "1.0.0-*",
  "buildOptions": {
    "debugType": "portable",
    "emitEntryPoint": true
  },
  "dependencies": {},
  "frameworks": {
    "netcoreapp1.1": {
      "dependencies": {
        "Microsoft.NETCore.App": {
          "type": "platform",
          "version": "1.1.0"
        }
      },
      "imports": "dnxcore50"
    }
  }
}
```

The code:
```cs
using System;

namespace ConsoleApplication
{
    public class Program
    {
        public static void Main(string[] args)
        {
            Console.WriteLine(GC.GetAllocatedBytesForCurrentThread());
        }
    }
}
```

I took a look at the System.Runtime.cs file and it's [using](https://github.com/dotnet/corefx/blob/master/src/System.Runtime/ref/System.Runtime.cs#L1252) the new moniker netcoreapp11 in few #ifdefs but none of these methods are exposed in System.Runtime. Moreover when I go to `.nuget\packages\System.Runtime\4.3.0\lib` I can see that there is no separate version for netcoreapp1.1. Anyway when I run the program an use `typeof(GC).GetTypeInfo().Assembly.Location` I get `C:\Program Files\dotnet\shared\Microsoft.NETCore.App\1.1.0\System.Private.CoreLib.ni.dll` which has the method exposed.

My question is: how can I use this method without reflection? Am I doing something wrong?

**Comment by karelz** [comment]
@ericstj @weshaggard @mellinoe can you please advise?

**Comment by weshaggard** [comment]
These new APIs didn't ship as part of .NET Core 1.1 release, they are targeting the .NET Core 1.2 release. There is some confusion as our master branch is still using the netcoreapp11 target framework because we haven't got the necessary nuget tool updates to start using netcoreapp1.2 yet. So don't take what you see in master targeting netcoreapp1.1 as a sign that it should have been in the .NET Core 1.1 release we just finished.

**Comment by karelz** [comment]
OK, that explains it: .NET Core 1.1 is in branch [release/1.1.0](https://github.com/dotnet/corefx/tree/release/1.1.0).

If you want to try out the newly added APIs, you need to grab master branch - some daily build.

**Comment by adamsitnik** [comment]
Thanks for the explanation, anyway I am not happy with the current situation.

One of the biggest advantages of .NET Core was supposed to be it's frequent releases. I know that I can get new package from custom feed, but I am building a framework (BenchmarkDotNet) and we can't ship a package to nuget.org with dependency to custom nuget feed ;(

**Comment by karelz** [comment]
Note that we have released quite a few versions in last few months -- 1.0, 1.0.1, 1.0.2 and 1.1 (see [.NET blog](https://blogs.msdn.microsoft.com/dotnet/)).

Currently we are working on 1.2, support .NET Standard 2.0 (see .NET Core [roadmap](https://blogs.msdn.microsoft.com/dotnet/2016/07/15/net-core-roadmap/)). Beside the fact that shipping official stable releases has some overhead, there is not easy way how to ship part of the new APIs and call them a meaningful set. .NET Standard 2.0 API set is THE meaningful set we came up with. Therefore shipping anything in the middle would be basically just a random set of APIs that happened to be finished by the time of shipping.
Also note that such 'random set of APIs' would have to have name, targeting pack, would require support, etc. -- if you add it all up, it's a lot of overhead, for questionable value.
Maybe in your case you just need that one single API, however, we have to think about all the customers and vast majority would not be happy with just that single API or a random snapshot of APIs.
